gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -271,6 +273,13 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  private boolean isNonDefaultServiceAccountAllowed() {
+    if (allowNonDefaultServiceAccount != null) {
+      return allowNonDefaultServiceAccount;
+    }
+    return credentials instanceof ComputeEngineCredentials;
```

Comment:
Do we need this? I see there is already the `isOnComputeEngine()` method, so ideally we want to keep compute-engine specific check in one place. Also, this method (`isNonDefaultServiceAccountAllowed()`) is used only in conjunction with `isOnComputeEngine()` on line 324 of this file).

Reply:
Sorry the name might be misleading:
1. `isOnComputeEngine()` checks the environment of the VM as DirectPath is allowed on Compute Engine/GKE but not App Engine. 
2. `isNonDefaultServiceAccountAllowed()` checks which service account the VM is using to access DirectPath. `credential` is instance of `ComputeEngineCredentials` guarantees that the VM is using the default service account (because the credential is retrieved from the metadata server), but it can not guarantee the environment (i.e., `ComputeEngineCredentials` can also be used on App Engine).
Based on the above understanding I think the two checks are different and should be kept.